### PR TITLE
created components for various types of components

### PR DESCRIPTION
**Pull request by micahcarroll**
#80

### CODE REVIEW

File: app/assets/javascripts/course/components/AudioComponent.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  onEnd() {
+    this.setState({ playing: false })
+    this.props.callback()
+  }
+
+  render() {
+
+    const {
+      playing
+    } = this.state
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/AudioComponent.js]: dont need this, just use this.state.playing.

File: app/assets/javascripts/course/components/AudioComponent.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import ReactPlayer from 'react-player'
+
+class AudioComponent extends React.Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {playing: true}
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/AudioComponent.js]: Add spaces between braces { hello: true }

File: app/assets/javascripts/course/components/CoursePage.js [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
+
+    this.component_params =  {type: 2, audio_url:'http://www.stephaniequinn.com/Music/Allegro%20from%20Duet%20in%20C%20Major.mp3', content_url: "ffDPTKn7HiY"}
+  }
+
   render() {
     return (
       <div>
         <h1>This is a course page</h1>
-        <CourseSidebar />
+        <CourseSidebar course_id={2} />
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/CoursePage.js]: remove placeholders.

File: app/assets/javascripts/course/components/FormComponent.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+
+class FormComponent extends React.Component {
+
+  constructor(props) {
+    super(props)
+
+    this.formUrl = this.props.form_url
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/FormComponent.js]: dont need this, just use this.props.form_url

File: app/assets/javascripts/course/components/ImageComponent.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import AudioComponent from './AudioComponent'
+
+
+class ImageComponent extends React.Component {
+
+  constructor(props) {
+    super(props)
+
+    this.imgUrl = this.props.imgUrl
+    this.audioUrl = this.props.audioUrl
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/ImageComponent.js]: same comment as above.

File: app/assets/javascripts/course/components/ParentComponent.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import ImageComponent from './ImageComponent'
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/ParentComponent.js]: space between React / ImageComponent

File: app/assets/javascripts/course/components/ParentComponent.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      case 2:
+        return <VideoComponent videoUrl={this.contentUrl} callback={this.onEnd} />
+        break;
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <p>This is a PARENT component</p>
```
Comment (charlesx2013) [inline app/assets/javascripts/course/components/ParentComponent.js]: Remove this.

**Comment by micahcarroll** [comment]
made the changes

**Comment by charlesx2013** [comment]
👍

**Pull request merged**